PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  SystemInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/29/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+#elseif os(watchOS)
+import UIKit
+import WatchKit
+#elseif os(macOS)
+import AppKit
+#endif
+
+public typealias RCSystemInfo = SystemInfo
+
+@objc(RCSystemInfo) open class SystemInfo: NSObject {
+
+    #if targetEnvironment(macCatalyst)
+    static let PLATFORM_HEADER = "uikitformac"
+    #elseif os(iOS)
+    static let PLATFORM_HEADER = "iOS"
+    #elseif os(watchOS)
+    static let PLATFORM_HEADER = "watchOS"
+    #elseif os(tvOS)
+    static let PLATFORM_HEADER = "tvOS"
+    #elseif os(macOS)
+    static let PLATFORM_HEADER = "macOS"
```

Comment:
maybe we should take the opportunity to update to lowerCamelCase here to follow Swift conventions

Reply:
Sounds good to me

Reply:
note: `platformHeader` is already used as a property, so I used `platformHeaderConstant`